homeassistant/components/saj/sensor.py

```diff
@@ -48,139 +58,161 @@
     {
         vol.Required(CONF_HOST): cv.string,
         vol.Optional(CONF_NAME): cv.string,
+        vol.Optional(CONF_SCAN_INTERVAL): cv.positive_int,
         vol.Optional(CONF_TYPE, default=INVERTER_TYPES[0]): vol.In(INVERTER_TYPES),
         vol.Inclusive(CONF_USERNAME, "credentials"): cv.string,
         vol.Inclusive(CONF_PASSWORD, "credentials"): cv.string,
     }
 )
 
 
-async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: Callable
+):
     """Set up the SAJ sensors."""
+    inverter: SAJInverter = hass.data[DOMAIN][entry.entry_id][DATA_INVERTER]
+    await inverter.setup(hass, async_add_entities)
 
-    remove_interval_update = None
-    wifi = config[CONF_TYPE] == INVERTER_TYPES[1]
-
-    # Init all sensors
-    sensor_def = pysaj.Sensors(wifi)
-
-    # Use all sensors by default
-    hass_sensors = []
-
-    kwargs = {}
-    if wifi:
-        kwargs["wifi"] = True
-        if config.get(CONF_USERNAME) and config.get(CONF_PASSWORD):
-            kwargs["username"] = config[CONF_USERNAME]
-            kwargs["password"] = config[CONF_PASSWORD]
-
-    try:
-        saj = pysaj.SAJ(config[CONF_HOST], **kwargs)
-        done = await saj.read(sensor_def)
-    except pysaj.UnauthorizedException:
-        _LOGGER.error("Username and/or password is wrong")
-        return
-    except pysaj.UnexpectedResponseException as err:
-        _LOGGER.error(
-            "Error in SAJ, please check host/ip address. Original error: %s", err
-        )
-        return
 
-    if not done:
-        raise PlatformNotReady
+async def async_setup_platform(
+    hass: HomeAssistant, config, async_add_entities: Callable, discovery_info=None
+):  # pragma: no cover
+    """Set up the SAJ inverter with yaml."""
+    _LOGGER.warning(
+        "Loading SAJ Solar inverter integration via yaml is deprecated. "
+        "Please remove it from your configuration"
+    )
+    inverter = SAJInverter(config)
+    await inverter.setup(hass, async_add_entities)
```

Comment:
We should not instantiate from config once the config flow has been in place. Instead, trigger the configuration flow to import the old YAML configuration into a config entry.